source/Octostache/Templates/Indexer.cs

```diff
@@ -26,7 +26,7 @@ public override string ToString()
             return "[" + (IsSymbol ? "#{"+ Symbol +"}" : Index) + "]";
         }
 
-        public override IEnumerable<string> GetArguments() => Symbol?.GetArguments() ?? new string[0];
+        public override IEnumerable<string> GetArguments() => Symbol?.GetArguments() ?? new string[]{ToString()};
```

Comment:
Could you please explain this change?

Reply:
This change is to enable the `IInputToken.GetArguments()` method to include the parsed `Indexer` value. I need this to be returned to assert the outcome of the new unit tests. I've looked into the usage of this method and this is only called in unit tests so I think this is a safe change.

Reply:
You are probably right about it just being used for testing, but just to understand the change, this basically (for the non `Symbol` case) changes from `new string[0]` to `$"[{Index}]"`
Is the Index always present if the Symbol is absent (it looks nullable so im guessing not), and if not should it fall back to the empty array result?

with the previous version of this code was it actually returning the empty string for a non symbol indexer?

Reply:
thanks @zentron, it's a good idea to fall back to `new string[0]` if the Index is also null.

Reply:
after discussing with RobW, I think we need to revert this change. This method is supposed to be internal and only used in tests. However, the class `TemplateToken` is public and so this method can still be called outside of `Octostache`. We found a usage here: https://github.com/OctopusDeploy/Calamari/blob/cbc46c67aa3b94e7d4b06a69c24694fb50bcb30e/source/Calamari/LaunchTools/NodeExecutor.cs#L73

As this was changed to enable some tests in `ParserFixture`, I will also remove those tests. I believe this is fine because we can rely on the tests in `UsageFixture` and `IterationFixture` to cover this change.